include/zephyr/net/net_l2.h

```diff
@@ -85,47 +85,31 @@ struct net_l2 {
 	extern const struct net_l2 NET_L2_GET_NAME(_name)
 #define NET_L2_GET_CTX_TYPE(_name) _name##_CTX_TYPE
 
-#ifdef CONFIG_NET_L2_VIRTUAL
 #define VIRTUAL_L2		VIRTUAL
 NET_L2_DECLARE_PUBLIC(VIRTUAL_L2);
-#endif /* CONFIG_NET_L2_DUMMY */
 
-#ifdef CONFIG_NET_L2_DUMMY
```

Comment:
I have been thinking of doing similar thing, thanks for sending this. There was some reason why the L2 defines were done like years ago but I fail to recall the reasoning, @tbursztyka do you remember why we hide the various L2 defines like this?